packages/core/src/utils-hoist/time.ts

```diff
@@ -67,26 +67,21 @@ function createUnixTimestampInSecondsFunc(): () => number {
 export const timestampInSeconds = createUnixTimestampInSecondsFunc();
 
 /**
- * Internal helper to store what is the source of browserPerformanceTimeOrigin below. For debugging only.
- *
- * @deprecated This variable will be removed in the next major version.
+ * Cached result of getBrowserTimeOrigin.
  */
-export let _browserPerformanceTimeOriginMode: string;
+let cachedTimeOrigin: [number | undefined, string] | undefined;
 
 /**
- * The number of milliseconds since the UNIX epoch. This value is only usable in a browser, and only when the
- * performance API is available.
+ * Gets the time origin and the mode used to determine it.
  */
-export const browserPerformanceTimeOrigin = ((): number | undefined => {
```

Comment:
I wonder how much of this logic we can simplify 🤔 

Probably not worth the effort.